### PR TITLE
feat: maintain element ordering in UI

### PR DESCRIPTION
**Pull request by bcoe**
- [x] still needs tests.

### CODE REVIEW

File: annotation.mustache [hunk 1]
```diff
@@ -3,22 +3,23 @@
     <li><h3>{{name}}</h3></li>
     {{#each rows}}
       <li>
-        {{#hasKey this "image"}}
-          <img src="{{{image.url}}}" alt="{{image.text}}" />
+      {{#each this}}
+        {{#hasKey this "_link_array"}}
```
Comment (nexdrew) [inline annotation.mustache]: I don't guess there's an "is array" check you can do in the template without the `_link_array` hackery?

**Comment by nexdrew** [comment]
So this is for maintaining order of elements within a row, now order of the rows themselves, correct?

File: index.js [hunk 2]
```diff
@@ -115,12 +117,26 @@ AnnotationPoller.prototype._applyReplacements = function (obj) {
       if (row.image) {
         if (row.image.url) row.image.url = _this._escape(row.image.url)
       }
+
+      // flatten the row object into an ordered
+      // set of elements. In our template we output
+      // the elements in the order that the keys appear.
+      Object.keys(row).forEach(function (key) {
```
Comment (nexdrew) [inline index.js:124]: Could we not just do the `Object.keys` iteration in the template?

**Comment by nexdrew** [comment]
This is somewhat unrelated, but what do you think about generating `annotation.js` as a `prepublish` step instead of maintaining it in source control?

File: index.js (hunk 2 above)
Comment (bcoe) [inline index.js:124]: Given the complexity of the transform, I'm inclined to do it outside the template. I'd like the implementers to continue using the existing format:

``` js
{
  rows: [
   {
      image: {},
      link: {}
   }
  ]
}
```

which then gets flattened into an array, and iterated over ... allowing the order of the keys in the object to have an effect on the order that elements are displayed in the addon.

**Comment by bcoe** [comment]
> This is somewhat unrelated, but what do you think about generating annotation.js as a prepublish step instead of maintaining it in source control?

done \o/

**Pull request merged**